src/polycap-description.c

```diff
@@ -0,0 +1,444 @@
+
+	return description;
+}
+
+//===========================================
+// get the polycap_profile from a polycap_description
+const polycap_profile* polycap_description_get_profile(polycap_description *description)
+{
+	return description->profile;
+}
+
+//===========================================
+// for a given array of energies, and a full polycap_description, get the transmission efficiencies.
+//   NOTE:
+// -Does not make photon image arrays (yet)
+// -in polycap-capil.c some leak and absorb counters are commented out (currently not monitored)
+// -Polarised dependant reflectivity and change in electric field vector missing
+int polycap_description_get_transmission_efficiencies(polycap_description *description, polycap_source *source, size_t n_energies, double *energies, double **efficiencies)
+{
+	int thread_cnt, thread_max, i, j;
+	int icount = 5000; //simulate 5000 photons hitting the detector
+	int64_t sum_istart=0, sum_irefl=0;
+	double *sum_weights;
+
+	// Check maximal amount of threads and let user choose the amount of threads to use
+	thread_max = omp_get_max_threads();
+	printf("Type in the amount of threads to use (max %d):\n",thread_max);
+	scanf("%d",&thread_cnt);
+	printf("%d threads out of %d selected.\n",thread_cnt, thread_max);
+
+	// Prepare arrays to save results
+	sum_weights = malloc(sizeof(double)*n_energies);
+	if(sum_weights == NULL){
+		printf("Could not allocate sum_weights memory.\n");
+		exit(1);
+	}
+	for(i=0; i<n_energies; i++) sum_weights[i] = 0.;
+
+//OpenMP loop
+#pragma omp parallel \
+	default(shared) \
+	private(i, j) \
+	num_threads(thread_cnt)
+{
+	int thread_id = omp_get_thread_num();
+	polycap_rng *rng;
+	unsigned int seed;
+	polycap_vector3 start_coords, start_direction, start_electric_vector;
+	double r; //random number
+	double n_shells; //amount of capillary shells in polycapillary
+	int boundary_check;
+	double src_rad_x, src_rad_y, phi; //distance from source centre in x and y direction and angle phi from x axis
+	double src_start_x, src_start_y;
+	double pc_rad, pc_x, pc_y; //pc radius and coordinates to direct photon to
+	polycap_photon *photon;
+	int iesc=-1, k;
+	int64_t istart=0; //amount of started photons
+	double *weights;
+
+	weights = malloc(sizeof(double)*n_energies);
+	if(weights == NULL){
+		printf("Could not allocate weights memory.\n");
+		exit(1);
+	}
+	for(k=0; k<n_energies; k++) weights[k] = 0.;
+
+	// Create new rng
+#ifdef _WIN32
+	rand_s(&seed);
+#else
+	FILE *random_device;
+	if((random_device = fopen("/dev/urandom", "r")) == NULL){
+		printf("Could not open /dev/urandom\n");
+		exit(2);
+	}
+	fread(&seed, sizeof(unsigned long int), 1, random_device);
+	fclose(random_device);
+#endif
+	rng = polycap_rng_new(seed);
+
+	i=0; //counter to monitor calculation proceeding
+	#pragma omp for nowait
+	for(j=0; j < icount; j++){
+		do{
+			// Obtain photon start coordinates
```

Comment:
Dit is een erg lang stuk voor je initieel foton te genereren.

Beter is een nieuwe functie te schrijven voor `polycap_source`:

```c
polycap_photon* polycap_source_get_photon(polycap_source *source, polycap_description *description, polycap_rng *rng);
```

Op die manier is de call naar `polycap_photon_new` niet meer nodig in deze functie.